gpytorch/lazy/lazy_tensor.py

```diff
@@ -899,7 +900,7 @@ def root_decomposition(self):
             batch_shape=self.batch_shape,
             matrix_shape=self.matrix_shape,
         )(*self.representation())
-        return res
+        return RootLazyTensor(res)
```

Comment:
In the old interpretation of `root_decomposition`, yes. My thought was to reinterpret `root_decomposition` now as "return a `RootLazyTensor` that approximates `self`. This is somewhat convenient because in several places the first thing we did after calling `root_decomposition` was create a `RootLazyTensor` anyways.